common/models/user.js

```diff
@@ -711,7 +780,17 @@ module.exports = function(User) {
         return cb(err);
       }
 
-      user.createAccessToken(ttl, function(err, accessToken) {
+      if (UserModel.settings.legacyPasswordFlow === false) {
+        const params = {
+          ttl: ttl,
+          scopes: ['reset-password'],
+        };
+        user.createAccessToken(params, onTokenCreated);
```

Comment:
See https://github.com/strongloop/loopback/pull/3350#issuecomment-293905818 and 745d20a